de.fhdo.lemma.eclipse.ui/src/de/fhdo/lemma/eclipse/ui/specify_url_dialog/SpecifyUrlDialog.xtend

```diff
@@ -0,0 +1,274 @@
+package de.fhdo.lemma.eclipse.ui.specify_url_dialog
+
+import org.eclipse.swt.widgets.Shell
+import org.eclipse.jface.dialogs.TitleAreaDialog
+import org.eclipse.jface.dialogs.MessageDialog
+import org.eclipse.swt.widgets.Composite
+import org.eclipse.jface.dialogs.IDialogConstants
+import org.eclipse.swt.graphics.Point
+import org.eclipse.swt.layout.GridLayout
+import org.eclipse.jface.dialogs.IMessageProvider
+import org.eclipse.swt.SWT
+import org.eclipse.swt.layout.GridData
+import org.eclipse.swt.widgets.Label
+import org.eclipse.swt.widgets.Text
+import org.eclipse.swt.widgets.Button
+import org.eclipse.swt.widgets.DirectoryDialog
+import org.eclipse.swt.events.SelectionListener
+import java.net.URL
+import java.net.MalformedURLException
+import org.eclipse.core.resources.ResourcesPlugin
+import de.fhdo.lemma.service.open_api.LemmaGenerator
+
+class SpecifyUrlDialog extends TitleAreaDialog {
+    static val MIN_DIALOG_WIDTH = 500
+    static val MIN_DIALOG_HEIGHT = 250
+    Text txtUrl
+
+    Text txtTargetLocation
+
+    Text txtDataModelName
+
+    Text txtServiceModelName
+
+    Text txtTechnologyModelName
+
+    public URL fetchUrl
```

Comment:
Is it necessary to have these fields public? If so, is it also necessary to enable other classes to write to them? If only read access is required, use Xtend's active `@Accessors` annotation (https://www.eclipse.org/xtend/documentation/204_activeannotations.html#property-annotation) to prevent write accesses.